src/main/java/com/formulasearchengine/mathmlquerygenerator/BasicXQueryGenerator.java

```diff
@@ -74,9 +74,14 @@ private String generateSimpleConstraints(Node node, boolean isRoot) {
                 //If an element node and not an attribute or text node, add to xquery and increment index
                 childElementIndex++;
 
-                // Only consider content mathml and ignore annotations or presentation mathml
-                if (!handleSpecialElements(child, childElementIndex) && (child.getLocalName() == null
-                        || !XMLHelper.ANNOTATION_XML_PATTERN.matcher(child.getLocalName()).matches())) {
+                // We only want to consider elements below the content mathml semantics, a deeper annotation-node will be ignored
+                boolean notAnnotationNode = child.getLocalName() == null || !XMLHelper.ANNOTATION_XML_PATTERN.matcher(child.getLocalName()).matches();
```

Comment:
maybe AnnotationNode and use ! in the if statement
and make handlespecialelements also a variable, I don't like running complex code inside of an if statement